### ISSUE TITLE
Cannot open playlist that includes songs that are not available

### ISSUE THREAD

**Issue by loveisfoss** [xou816/spot#184]
**Describe the bug**
When opening a playlist that includes content that is not available for playing, the playlist's song are not displayed at all. A good example is when a playlist includes local content and one tries to access it while the PC that was used as a "server" for these local files is currently offline/shut down.

**To Reproduce**
1. Go to Playlists
2. Click on a playlist that includes a song that is unavailable

**Expected behavior**
When opening a playlist that includes a file that is unavailable: Show the playlist songs and just gray out the file that cannot be played.

**Screenshots**
![Screenshot from 2021-04-02 10-50-59](https://user-images.githubusercontent.com/51087561/113399945-57622680-93a1-11eb-8c9a-1b1baebf8e71.png)

**General information:**
 - Distribution: Debian bullseye
 - Installation method: Flathub
 - Version: 0.1.12
 - Device used: Desktop

**Stack trace:**
flatpak run --env=RUST_BACKTRACE=full dev.alextren.Spot
```
(process:2): Gtk-WARNING **: 10:47:26.973: Locale not supported by C library.
	Using the fallback 'C' locale.
home
bitrate: Bitrate160
using pulseaudio
playlist_xxxxxxxxxxx
Error: ParseError(Error("invalid type: null, expected a string", line: 3089, column: 19))
playlist_yyyyyyyyyyy
Error: ParseError(Error("invalid type: null, expected a string", line: 545, column: 19))

```

**Comment by xou816** [comment]
Hi! Thanks for the report. I thought we had handled this previously :( 
Do you have an example of a public playlist with one such track? Or an excerpt of the network response (for instance, the line being pointed at, or the object representing one such track)? 
If needed, cached files for the network responses are in .var/app/dev.alextren.Spot/cache/spot/net.

**Comment by loveisfoss** [comment]
You are right! I can open on Spot a public playlist that has an unavailable song. I can see that the song is unavailable on Spotify. Spot doesn't gray it out and lists it normally. When trying to play it, it just skips to the next track. I think this is the expected behavior as of now.

So, the problem seems to be related with my own playlists that include local-only songs. I have added for example a local mp3 file in my playlist and I have done it with the Spotify client on windows. It is this playlist that I cannot open. I am also going to try having Spotify on Windows open when trying to open the playlist on Spot and come back with feedback.

Is it something special you want to see on the cached files?

**Comment by xou816** [comment]
> You are right! I can open on Spot a public playlist that has an unavailable song. I can see that the song is unavailable on Spotify. Spot doesn't gray it out and lists it normally. When trying to play it, it just skips to the next track. I think this is the expected behavior as of now.

Better than an error or a crash, but still could be improved!

> So, the problem seems to be related with my own playlists that include local-only songs. I have added for example a local mp3 file in my playlist and I have done it with the Spotify client on windows. It is this playlist that I cannot open on Spot. I am also going to try having Spotify on Windows open when trying to open the playlist on Spot and come back with feedback.

Okay, I'll try this :) I'll get back to you if I can't work it out from here!

**Comment by LennyPenny** [comment]
here is one of my public playlists that has this exact problem in spot: https://open.spotify.com/playlist/5FKAf83T8zZHBDu2xEgPNg?si=MRv2XhQESvy8q8MhdaMZRw

**Comment by LennyPenny** [comment]
here is a paste of the json data for convience: https://snippet.host/kvbd
it fails at the first track where `is_local` is true and it seem that the `album.id` can't be null in spot (probably more as well)

**Comment by xou816** [comment]
Thanks, I could reproduce it with your playlist! Indeed, the track object that causes the error is missing most fields and is messing up the parsing. I played a bit with [untagged enums](https://serde.rs/enum-representations.html#untagged) in serde, looks like it should help with solving this!

**Comment by xou816** [comment]
fixed in 0.1.13
